### ISSUE TITLE
grub-bhyve hangs with 100% CPU usage when booting Debian

### ISSUE THREAD

**Issue by fjerhammer** [pr1ntf/iohyve#112]
I cannot get Debian to boot with neither `os=debian` nor `os=custom`. It worked fine during `iohyve install` (using `debian-8.3.0-amd64-CD-1.iso`), but after destroying it and doing the initial `iohyve start` it spawns a grub-bhyve process that takes up 100% CPU, freezing cu and making the VM unable to start.
Am I doing something wrong? I have tried reinstalling GRUB using `grub-install /dev/sda` in the VM, to no avail. `/boot` is on its own partition (#1).

I learned about `os=custom` from the wiki page on CentOS, even though the feature doesn't seem to be documented in the manpage. I tried the following:

```
iohyve set web os=custom
iohyve set web autogrub='linux%1s(hd0,msdos1)/vmlinuz-3.16.0-4-amd64%1sroot=/dev/sda3%1sro%1squiet\ninitrd%1s(hd0,msdos1)/initrd.img-3.16.0-4-amd64\nboot\n'
```

However, the result was exactly the same after starting it.
I was wondering if the GRUB string might get cut off prematurely due to a character limit on ZFS property values:

```
# iohyve get web autogrub
Getting web prop autogrub...
linux%1s(hd0,msdos1)/vmlinuz-3.16.0-4-amd64%1sroot
# zfs get iohyve:autogrub zroot/iohyve/web
NAME              PROPERTY         VALUE                                               SOURCE
zroot/iohyve/web  iohyve:autogrub  linux%1s(hd0,msdos1)/vmlinuz-3.16.0-4-amd64%1sroot  local
```

Although it can just as well be a limit on the output of `zfs get`.
If I boot the VM by manually running `grub-bhyve` with the above boot parameters, it starts just fine.

I am using FreeBSD 10.2-RELEASE-p12 with iohyve-0.7.3 from ports.

**Comment by pr1ntf** [comment]
First off, my apologies for the radio silence for a week. 

Secondly, thanks for the detailed report! I do have a few questions. 

Did you do an LVM install of Debian? If so, you may want to try `os=d8lvm`.

If you did a standard install of Debian without LVM support, can you try `os=default`?

As far as `autogrub` goes, it's very hacky and can lead to various issues. Because of this, we are phasing out the feature in favor of custom `grub.cfg` files in [iohyve](https://github.com/pr1ntf/iohyve/wiki/General-Path-to-1.0#075-goals-submit-to-ports).

Thanks!

**Comment by pr1ntf** [comment]
I'm going to go ahead and close this issue. If you have further issues related to this, feel free to pop back into this thread and I'll reopen.

**Comment by ohmer1** [comment]
Hello,

I'm using v0.7.7 on FreeNAS. I think I have the same problem.

grub-iohyve taking 100% cpu and iohyve start saying
`GRUB Process does not run in background....
If your terminal appears to be hanging, check iohyve console backupqc in second terminal to complete GRUB process...`

I installed an Ubuntu 16.04.1 VM (standard partition, no LVM) but once the VM is installed, I can't boot it. Setting os=default doesn't change anything. I already manually patched the issue fixed in a95e1d2.

I just reinstalled my VM with LVM this time (and os=d8lvm) and the VM is now booting.

**Comment by pr1ntf** [comment]
I'm not sure if that fix made it into the latest release or not. If not we can tag a new one soon™️. In the mean time, I would suggest upgrading to iohyve version 0.7.8 as it came with some more FreeNAS related bug fixes. If that is not the latest version provided by FreeNAS ports, I suggest pinging someone on the FreeNAS forums like their perpetually busy engineer Kris. 

If you continue to have issues related to this, please feel free to open a new ticket. Thanks!
